### ISSUE TITLE
Netlify LFS Media Manager incorrectly assembles image urls on localhost

### ISSUE THREAD

**Issue by ryanfiller** [netlify/netlify-cms#3117]
**Describe the bug**
When using Netlify Large Media, files are hosted on the master branch of the repository at the url specified in the CMS config `public_folder`. However, when running the CMS locally, the Media Manager is looking for images hosted on `localhost`. 

**To Reproduce**
Any setup as outline above should reproduce the issue. My project can be cloned here - https://github.com/ryanfiller/portfolio-gatsby-v2

**Expected behavior**
It would be great if the CMS config `site_url` could be used in instances of git LFS to look to the deployed images on production to thumbnail them locally.

**Screenshots**
<img width="1169" alt="Screen Shot 2020-01-20 at 9 39 30 AM" src="https://user-images.githubusercontent.com/19308155/72739379-26fde600-3b69-11ea-9eef-a8816b917bb9.png">

**Applicable Versions:**
 - Netlify CMS version: "netlify-cms-app": "^2.11.1",
 - Git provider: GitHub
 - OS: macOS
 - Browser version: Firefox 72.0.1

**CMS configuration**
https://github.com/ryanfiller/portfolio-gatsby-v2/blob/master/src/cms/cms.js

**Additional context**
This is not an urgent issue. The media manager functions correctly and is usable by looking at the file names, any client using the CMS would probably be doing so in production rather than on localhost. This is simply a QOL/DX request/bug.

**Comment by ryanfiller** [comment]
This issue doesn't seem too terribly complicated on the surface, I'm interested in fixing it myself and creating a PR if someone could help and point me in the right direction.

**Comment by erezrokah** [comment]
Might be related https://github.com/netlify/netlify-cms/issues/1206#issuecomment-569706509

**Comment by ryanfiller** [comment]
@erezrokah Yes, this looks like its the same issue. I think the fix is to have the media manager always look to master/the deployed branch?

**Comment by erezrokah** [comment]
It should work that way - always get the data from the current branch. The code that loads the images is here: https://github.com/netlify/netlify-cms/blob/0755f90142525d94158c1d112a5549e21eef77c2/packages/netlify-cms-backend-git-gateway/src/implementation.ts#L434 is you want to try to debug it

**Comment by ryanfiller** [comment]
So digging in, this seems a little more complicated than it did at first.
I don't actually have `netlify-cms-backend-git-gateway` as a requirement in my project, I have `netlify-cms-app`, which in turn requires `"netlify-cms-backend-git-gateway": "^2.6.0"`.

The monorepo is confusing me a little also - it seems like 2.6.0 is javascript but 2.6.2 has been converted over to typescript?

**Comment by erezrokah** [comment]
> The monorepo is confusing me a little also - it seems like 2.6.0 is javascript but 2.6.2 (most current version on npm) has been converted over to typescript?

We've started converting parts of the repo to TypeScript.

**Comment by zackseuberling** [comment]
>  This is not an urgent issue. The media manager functions correctly and is usable by looking at the file names, any client using the CMS would probably be doing so in production rather than on localhost. This is simply a QOL/DX request/bug.

I'm running into similar issues in a production environment using 11ty and Netlify:

<img width="718" alt="image" src="https://user-images.githubusercontent.com/1609558/87985175-feab6180-ca8f-11ea-857b-2c580c70c5b9.png">

Maybe my site isn't configured correctly, but I can't find anything in the documentation that mentions this

**Comment by martinjagodic** [comment]
Closing as not planned because Netlify Large Media is sadly deprecated

**Comment by ryanfiller** [comment]
> Closing as not planned because Netlify Large Media is sadly deprecated

Deprecated from the CMS or not offered at all by Netlify anymore? This is the first I've heard of this.

**Comment by martinjagodic** [comment]
[Deprecated by Netlify](https://answers.netlify.com/t/large-media-feature-deprecated-but-not-removed/100804) for almost a year now. At Decap, we try to narrow our limited resources to future-proof features, so supporting NLM doesn't fall in that category. This is a big blow because many of us relied on that feature, but it's beyond our control.

**Comment by ryanfiller** [comment]
Gotcha, thank you for the reply and I see their blog post on the situation now. And yeah... bummer, I guess I need to replace this with something else now.
